onnx/defs/training/defs.cc

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) ONNX Project Contributors.
+// Licensed under the MIT license.
+
+#include <algorithm>
+#include <cmath>
+#include "onnx/defs/function.h"
+#include "onnx/defs/schema.h"
+
+namespace ONNX_NAMESPACE {
+static const char* Gradient_ver12_doc = R"DOC(
+Gradient operator computes the derivatives of some tensors with respect to
```

Comment:
It might be worth adding an explanatory note somewhere, something like "The gradient operator is slightly different from the standard operators. It is conceptually a higher-order operator that takes a computation-graph as an attribute, but we abuse the attribute mechanism to describe the computation-sub-graph implicitly by listing the inputs and output of the computation-sub-graph in the current graph.'